plugin/trino-hive/src/main/java/io/trino/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -385,11 +389,13 @@ private ListenableFuture<Void> loadPartition(HivePartitionMetadata partition)
         InputFormat<?, ?> inputFormat = getInputFormat(configuration, schema, false);
         FileSystem fs = hdfsEnvironment.getFileSystem(hdfsContext, path);
         boolean s3SelectPushdownEnabled = shouldEnablePushdownForTable(session, table, path.toString(), partition.getPartition());
-
-        // S3 Select pushdown works at the granularity of individual S3 objects,
-        // therefore we must not split files when it is enabled.
+        // S3 Select pushdown works at the granularity of individual S3 objects for compressed files
+        // and finer granularity for uncompressed files using scan range feature.
+        // Scan range is currently enabled for CSV.
+        boolean isCsv = CSV.equals(getDataType(getDeserializerClassName(schema)).orElse(null));
```

Comment:
Will be nice to enable this feature for JSON as a follow up PR.

Reply:
Thanks for raising this. That's the plan, as I also mentioned in the description :D